### PR TITLE
WT-6981 Add randomness to Python test suite runs

### PR DESCRIPTION
**Pull request by boboeee**
The purpose of this ticket is to investigate the possibility of adding more randomness to the python test suite. This can be done through giving the option to randomise the seed for random numbers. This achieves more test coverage through generating unpredictable random numbers through the option of **-R**,  and the option of running two specific random integer seeds is now available through **-S**

The reason for the generating the random seed from 1-2^32, is to make sure that the seed doesn't get 0, which isn't allowed in the formula for generating random numbers

The random seed will only be used in a test case when the "@wttest.randomseed()" is applied to the test case, therefore using the random seed for specific tests.

### CODE REVIEW

File: test/suite/run.py [hunk 1]
```diff
@@ -415,6 +418,17 @@ def error(exitval, prefix, msg):
...
+            if option == '-randomseed' or option == 'R':
+                seedw = random.randint(1, 0xffffffff)
+                seedz = random.randint(1, 0xffffffff)
+                continue
+            if option == '-seed' or option == 'S':
+                if seedw != 0 or seedz != 0 or len(args) < 2:
+                    usage()
+                    sys.exit(2)
+                seedw = args.pop(0)
+                seedz = args.pop(0)
```
Comment (ddanderson) [inline test/suite/run.py:430]: Should check to make sure that neither seedw and seedz are not 0.  It shouldn't be possible, and will probably cause parts of the code to break.

File: test/suite/wttest.py [hunk 2]
```diff
@@ -766,6 +773,8 @@ def runsuite(suite, parallel):
         WiredTigerTestCase._concurrent = True
         suite_to_run = ConcurrentTestSuite(suite, fork_for_tests(parallel))
     try:
+        if WiredTigerTestCase._randomseed:
+            WiredTigerTestCase.prout("Starting test suite with seedw=" + str(WiredTigerTestCase._seeds[0]) + " and seedz=" + str(WiredTigerTestCase._seeds[1]))
```
Comment (ddanderson) [inline test/suite/wttest.py:777]: Assuming you combine the seeds suggested above, I would say: `Rerun this test or any part of it with "--seed wwwww.zzzzzz" to get the same randomness` .  That gives the user clear instructions on what to do with the information.

File: test/suite/wttest.py [hunk 3]
```diff
@@ -221,6 +221,11 @@ def globalSetup(preserveFiles = False, useTimestamp = False,
         WiredTigerTestCase._concurrent = False
         WiredTigerTestCase._globalSetup = True
         WiredTigerTestCase._ttyDescriptor = None
+        WiredTigerTestCase._seeds = [521288629, 362436069]
```
Comment (ddanderson) [inline test/suite/wttest.py:224]: I like that your changes have the "standard" random seeds in only one occurrence now.

File: test/evergreen.yml [hunk 4]
```diff
@@ -1447,6 +1447,15 @@ tasks:
...
 
+  # Run the tests that uses suite_random with a random starting seed
+  - name: unit-test-random-seed
+    depends_on:
+    - name: compile
+    commands:
+      - func: "fetch artifacts"
+      - func: "unit test"
+        vars:
+          unit_test_args: -v 2 -R cursor13 join02 join07 schema03 timestamp22
```
Comment (ddanderson) [inline test/evergreen.yml:1458]: I think this is fine for now, but we should think about how to keep this list up to date.

**Review by ddanderson — APPROVED** [review]
I like this change overall.  I suggested a couple mods that I think would improve how it is used.

File: test/suite/wttest.py (hunk 2 above)
Comment (boboeee) [reply test/suite/wttest.py:777]: done!

File: test/suite/wttest.py (hunk 3 above)
Comment (boboeee) [reply test/suite/wttest.py:224]: thank you!

File: test/suite/run.py (hunk 1 above)
Comment (boboeee) [reply test/suite/run.py:430]: done!

File: test/suite/run.py [hunk 5]
```diff
@@ -124,6 +124,8 @@ def usage():
   -t      | --timestamp          name WT_TEST according to timestamp\n\
   -v N    | --verbose N          set verboseness to N (0<=N<=3, default=1)\n\
   -i      | --ignore-stdout      dont fail on unexpected stdout or stderr\n\
+  -R      | --randomseed         run with random seeds for generating random numbers\n\
+  -S      | --seed               run with specified two specified seeds for generating random numbers\n\
```
[the remark this answers is not in the document]
Comment (boboeee) [reply test/suite/run.py:128]: like that idea, implemented!

File: test/evergreen.yml (hunk 4 above)
Comment (boboeee) [reply test/evergreen.yml:1458]: true, should I create a follow up ticket for this @ddanderson ?

Comment (tammybailey) [reply test/evergreen.yml:1458]: @boboeee yes, please create a follow up ticket and link it to this one. thanks!

**Review by tammybailey — APPROVED** [review]
LGTM!

**Pull request merged**